python/tvm/tir/transform/inject_rolling_buffer.py

```diff
@@ -0,0 +1,214 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Inject rolling buffers through a TIR transformation."""
+# pylint: disable=invalid-name,unused-argument,inconsistent-return-statements
+from collections import defaultdict
+
+import tvm
+from tvm import arith
+
+
+def InjectRollingBuffer():
```

Comment:
I think generally it would better to give some extensive documentation -- possibly with an example two TIR snippets to show the transformation. 
As a community, we should try to encourage this.

Reply:
The necessary TIR snippets would be very large for this particular pass. I have a before/after TIR example in the tests so hopefully these could be used as a reference. In lieu of TIR in the documentation, I've improved it somewhat and included a link to the RFC which has a far more detailed explanation of the pass + diagrams. Let me know if you think this is sufficient.

Reply:
Fair enough :)